test/attr/attr_availability_transitive_osx.swift

```diff
@@ -35,3 +39,70 @@ func osx_extension_call_osx_extension() {
 func osx_extension_call_osx() {
     osx() // expected-error {{'osx()' is unavailable}}
 }
+
+@available(OSX, unavailable)
+var osx_init_osx = osx() // OK
+
+@available(OSXApplicationExtension, unavailable)
+var osx_extension_init_osx = osx() // expected-error {{'osx()' is unavailable}}
+
+@available(OSX, unavailable)
+var osx_inner_init_osx = { let inner_var = osx() } // OK
+
+// FIXME: I'm not sure why this produces two errors instead of just one.
```

Comment:
I think it's because someone is not skipping multi-statement closures somewhere, so the body of the closure is visited again. Just a guess though.